### ISSUE TITLE
RealmDAO

### ISSUE THREAD

**Issue by mechaquan** [TaggrNetwork/taggr#119]
Create a realm based SNS alternative solution for DAO launches

High level thoughts are n $Taggr to launch (autonomously either with or without a TaggrDAO gating vote) and realm is launched with $token (allocation determined by realm controller and clearly laid out in realm white paper).  RealmDAO will have a copy/paste of TaggrDAO binary voting functionality and can choose if it wants TaggrDAO as a controller or not (maybe NNS is there as default).

Thoughts?

**Comment by mechaquan** [comment]
Yeah I was kinda thinking it would also drive some additional development efforts from teams wanting the DAO functionality... Taggr doesn't have to promise anything.

I'm also torn on gating but I don't want Taggr to turn into Memeland either...

It was kinda a crazy idea but I do think it would drive adoption!

**Comment by mechaquan** [comment]
I just want to add that there is currently nothing available for small teams not wanting to go the SNS route.  There is a huge opportunity here.

I'm already talking to a team who has been around the ecosystem since the beginning and they are very interested in something like this.

We could attract dozens of devs to Taggr and as a knock on effect users.

Taggr could become the townhall of DAOs.

Wrt to the gating / paying to launch... I feel it's gotta be one or the other because just like we've seen with realms we can't just have 1000s of pointless DAOs... if it cost 1000 $Taggr to launch then at least the value will naturally gate it

What would the bare minimum simplest implementation look like to provide a solution for this?  I'm just curious to scope it... if it's not feasible then we can just move on...  

In my simple mind, a RealmDAO launch could be spinning up a copy of the proposal voting DAO structure in a different canister with the Taggr Realm post functionality and then the RealmDAO would add this canister as the controller to the dapp canister for which it would push binary updates to.  I don't think these RealmDAOs should earn $Taggr for posts within that structure but they could "broadcast" to Taggr which would give them the ability to earn $Taggr.

I appreciate this isn't something simple to implement and I'd say it would require a technical team to help support the POC...

**Comment by taggrx** [comment]
Why are we creating proposals and discussing them here and not on Taggr?! 😅

Regarding the actual idea. I think I totally agree with Nathan. Yes, it would be a massive development effort, but the most important question to me if it's strategically meaningful? I firmly believe in simple tools doing one job exceptionally well and being composable with other tools like lego bricks. Blowing up Taggr to an SNS alternative is going far away from a social network.

If there is a gap in certain functionality for DAOs on IC it would make sense to form a working group with all those who need a solution and then implement something tailored to their concrete needs. Taggr could be pluggable into it, but growing this inside Taggr - I am really sceptical about this.

**Comment by mechaquan** [comment]
I brought it up here because god forbid it gets supported in a poll and you don't implement! 😂

**Comment by mechaquan** [comment]
> Yes, it would be a massive development effort, but the most important question to me if it's strategically meaningful? I firmly believe in simple tools doing one job exceptionally well and being composable with other tools like lego bricks. Blowing up Taggr to an SNS alternative is going far away from a social network.
> 
> If there is a gap in certain functionality for DAOs on IC it would make sense to form a working group with all those who need a solution and then implement something tailored to their concrete needs. Taggr could be pluggable into it, but growing this inside Taggr - I am really sceptical about this.

I want to come back to this... not because I want to push for Taggr becoming an SNS alternative but because I think communication is paramount to any DAO and I think that is the foundation of any DAO... being able to communicate on a truly decentralized communication platform.

Also, I don't suggest going from 0 to 100 on this idea but rather taking small steps that add value and utility to realms.  If we can attract teams to come to Taggr then it can only help, right?

One small idea would be to put some more economics around realms... maybe allow them to have their own tokens for communications within that realm (so we don't end up with a group just talking in a realm and karma farming and we just don't see it because we're only using followed) and also allow them to earn some platform revenue generated by that realm.  I mean this would be better than the alternative social "dapps" who don't give the users any revenue or reward them for actually making the platform successful.

To me these are little baby steps to a full blown realmDAO... and we don't have to go all the way but I think we need a long term vision or strategy and then just travel in that direction.

**Comment by taggrx** [comment]
This could be interesting in a hypothetical future where we have thousands of realms full of activity with Taggr being an established decentralized social platform, with hundreds of DAOs on IC...

But right now, with 80 users logging into the entire Taggr per _week_, I don't know where to find motivation to work on features for this highly hypothetical future. Do you really think that once these features are implemented, this demand will somehow materialize in form of new users forming one DAO after another one?

Idk maybe it's just me lacking the vision or the entrepreneurial spirit, this is why it would be helpful to have more inputs and discuss this on Taggr itself...

**Comment by mechaquan** [comment]
I will open up a more generic discussion but to answer your question below

> Do you really think that once these features are implemented, this demand will somehow materialize in form of new users forming one DAO after another one?

It's just one idea on how to add utility and value to the platform... because let's be honest they aren't coming to Taggr to get content they can't get elsewhere.  So my thoughts were to focus and address needs/utility that would benefit our immediate target audience of ICP users.
